### DISCUSSION TITLE
How to force SLBE system to use NextDNS with systemd-resolved?

### DISCUSSION THREAD

**Discussion by birds-swim** [SpiralLinux/SpiralLinux-project#466] [Q&A] [answered]
Hello.

I am trying to get my fresh install of Spiral Linux Builder Edition to use NextDNS as my primary DNS resolver, but it seems like SL is ignoring my `/etc/systemd/resolved.conf`.

NextDNS offers a installation script, but I don't want to use it because it messes with my apt sources to create a ".deb" package. Kind of a barbaric way to force it to work, and to their credit, it gets the job done. 

But I really don't want to use the script. I want to use the tools that were shipped with Debian/SLBE.

How do I get SLBE to use NextDNS as my DNS resolver and make sure my system doesn't use anything else? Is SL even using systemd-resolved? Or some other program?

**Comment by geckolinux** [comment, marked as answer]
Hi there, the Builder edition doesn't come with `systemd-resolved` by default. I would recommend sticking with NetworkManager for that. If you check the `/etc/resolv.conf` file you should see the first line `# Generated by NetworkManager`, and if that's the case you can use whichever NetworkManager GUI you installed for your desktop environment to set a different DNS server. There's even a console GUI config tool for NetworkManager in the Builder edition called `nmtui` that you could use.

Just a suggestion too: The best place to configure the DNS server is usually on your router, and then it will get served out to all connected devices automatically via DHCP and won't require manual configuration on any of the devices.

**Reply by birds-swim** [reply]
Thanks Geck for your reply.

I'll test out your suggestions later today.

"Just a suggestion too: The best place to configure the DNS server is usually on your router, and then it will get served out to all connected devices automatically via DHCP and won't require manual configuration on any of the devices."

Yeah, this is better. But my family members won't let me touch their router, so I'm left with configuring this on my system.

**Reply by birds-swim** [reply]
Sweet! This worked! Now I gotta figure out how to encrypt my DNS with NetworkManager.

**Reply by geckolinux** [reply]
It doesn't look like NetworkManager supports DoH or DoT. But you can install `systemd-resolved` and configure it there. I generally understand the concept of how to do so and how the pieces fit together with these three links:

1. https://mullvad.net/en/help/dns-over-https-and-dns-over-tls#linux
2. https://shivering-isles.com/2021/01/configure-dot-on-systemd-resolved
3. https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/258
